homeassistant/components/vesync/light.py

```diff
@@ -82,3 +87,36 @@ def supported_features(self):
     def brightness(self):
         """Get dimmer brightness."""
         return round((int(self.dimmer.brightness) / 100) * 255)
+
+
+class VeSyncBulbHA(VeSyncDevice, LightEntity):
+    """Representation of a VeSync bulb."""
+
+    def __init__(self, bulb):
+        """Initialize the VeSync bulb device."""
+        super().__init__(bulb)
+        self.bulb = bulb
+
+    def turn_on(self, **kwargs):
+        """Turn the device on."""
+        if ATTR_BRIGHTNESS in kwargs:
+            # get brightness from HA data
+            brightness = int(kwargs[ATTR_BRIGHTNESS])
+            # convert to percent that vesync api expects
+            brightness = round((brightness / 255) * 100)
+            # clamp to 1-100
+            brightness = max(1, min(brightness, 100))
+            self.bulb.set_brightness(brightness)
+        # Avoid turning device back on if this is just a brightness adjustment
+        if not self.is_on:
+            self.device.turn_on()
+
+    @property
+    def supported_features(self):
+        """Get supported features for this entity."""
+        return SUPPORT_BRIGHTNESS
+
+    @property
+    def brightness(self):
+        """Get bulb brightness."""
+        return round((int(self.bulb.brightness) / 100) * 255)
```

Comment:
Does this do I/O?

Reply:
By I/O, you mean talk to the API library?  I believe it does, I mimicked the other parts of this integration for this part. it worked on my tests.

Reply:
It looks like its accessing a cached property and doesn't do any I/O when calling .brightness

https://github.com/webdjoe/pyvesync/blob/master/src/pyvesync/vesyncbulb.py#L37

This should be fine